sphinxcontrib/confluencebuilder/builder.py

```diff
@@ -12,6 +12,7 @@
 from docutils import nodes
 from docutils.io import StringOutput
 from os import path
+from jinja2 import Environment, BaseLoader
```

Comment:
A potential recommendation, if I may, is that maybe we do not need to explicitly import/reference the jinja2 engine. Sphinx already provides [some glue](https://github.com/sphinx-doc/sphinx/blob/master/sphinx/jinja2glue.py) to utilize jinja2 in Sphinx, and Confluence builder [should be binding a jinja2 environment/base loader automatically](https://github.com/sphinx-contrib/confluencebuilder/blob/master/sphinxcontrib/confluencebuilder/builder.py#L133-L134). An example case we have in this extension is to [utilize template processing](https://github.com/sphinx-contrib/confluencebuilder/blob/fe503c0f4619f80febfd0a0c362ba5c45c5c41c3/sphinxcontrib/confluencebuilder/storage/index.py#L52-L58) for a configured index page.

Maybe, instead of:

```
rtemplate = Environment(loader=BaseLoader).from_string(template_data)
self._cached_header_data = rtemplate.render(**self.config.confluence_header_data)
```

Something like this would work (note: not tested this):

```
output = self.templates.render_string(template_data.decode('utf-8'),
    self.config.confluence_header_data)
```

If we can do this, it would could:

- Drop the need for these new imports.
- Drop the need to add the `jinja2` package added to `requirements_validation.txt`.

Reply:
I've done this too, the decode didn't seem to be necessary, and in the translator I had to grab the templates object from the builder, but it is certainly much cleaner without that extra dependency!